// @wonderflow/react-components/dist/components/snackbar/snackbar.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/*
 * Copyright 2022-2023 Wonderflow Design Team
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import clsx from 'clsx';
import { forwardRef } from 'react';
import { Button, Stack, Symbol, Text, } from '../..';
import * as styles from './snackbar.module.css';
const mapping = {
    info: 'informative',
    warning: 'warning',
    neutral: 'neutral',
    positive: 'positive',
    danger: 'danger',
};
export const Snackbar = forwardRef(({ children, className, title, icon, kind = 'neutral', as: Wrapper = 'output', dismissable, dismissLabel = 'Dismiss', onDismiss, ...otherProps }, forwardedRef) => {
    const defaultIcons = {
        info: 'circle-info',
        warning: 'triangle-exclamation',
        neutral: 'compass',
        positive: 'circle-check',
        danger: 'circle-x',
    };
    return (_jsx(Wrapper, { ref: forwardedRef, className: clsx(styles.Snackbar, className), "data-snackbar-kind": kind, role: "status", ...otherProps, children: _jsxs(Stack, { vAlign: "start", hAlign: "start", direction: "row", columnGap: 16, children: [_jsx(Symbol, { "data-snackbar-has-title": !!title, className: styles.Icon, weight: "outline", source: icon ?? defaultIcons[kind], dimension: 24 }), _jsxs(Stack, { rowGap: 16, children: [_jsxs(Stack, { rowGap: 4, children: [title && _jsx(Text, { variant: "subtitle-1", color: kind ? mapping[kind] : 'neutral', preventResponsive: true, children: title }), _jsx(Text, { variant: "body-2", children: children })] }), dismissable && (_jsx(Stack, { hAlign: "end", children: _jsx(Button, { onClick: onDismiss, className: styles.Action, dimension: "small", children: dismissLabel }) }))] })] }) }));
});
