// @wonderflow/react-components/dist/components/card/card.stories.js
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import { Symbol, Text, } from '../..';
import { Card } from './card';
const story = {
    title: 'Layouts/Card',
    component: Card,
    args: {
        bordered: false,
        vibrant: false,
        highlightOnHover: true,
        dimmed: 1,
        children: (_jsxs(_Fragment, { children: [_jsx(Text, { variant: "heading-4", children: "Card title" }), _jsx(Text, { children: "Lorem, ipsum dolor sit amet consectetur adipisicing elit." })] })),
    },
    argTypes: {
        children: {
            table: {
                disable: true,
            },
        },
    },
};
export default story;
const Template = args => _jsx(Card, { ...args });
export const Default = Template.bind({});
Default.args = {};
export const Vibrant = Template.bind({});
Vibrant.args = {
    vibrant: true,
};
export const WithLeft = Template.bind({});
WithLeft.args = {
    left: _jsx(Symbol, { source: "bell", dimension: 32 }),
};
export const WithRight = Template.bind({});
WithRight.args = {
    right: _jsx(Symbol, { source: "bell", dimension: 32 }),
};
export const WithLeftAndRight = Template.bind({});
WithLeftAndRight.args = {
    left: _jsx(Symbol, { source: "bell", dimension: 32 }),
    right: _jsx(Symbol, { source: "calendar", dimension: 32 }),
};
